src/app/InteractionModelEngine.cpp

```diff
@@ -2032,6 +2032,54 @@ size_t InteractionModelEngine::GetNumDirtySubscriptions() const
     return numDirtySubscriptions;
 }
 
+#if CHIP_CONFIG_ENABLE_ICD_SERVER && CHIP_CONFIG_ENABLE_ICD_CIP
+bool InteractionModelEngine::ShouldCheckInMsgsBeSent(FabricIndex aFabricIndex, NodeId subjectID)
+{
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+    // If at least one registration has a persisted entry, do not send Check-In message.
+    // The resumption of the persisted subscription will serve the same function a check-in would have served.
+    return !app::InteractionModelEngine::GetInstance()->SubjectHasPersistedSubscription(aFabricIndex, subjectID);
+#else
+    return true;
+#endif // CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+}
+
+void InteractionModelEngine::ScheduledForceTriggerCheckInMessages()
+{
+    SessionManager * sessionManager = mpExchangeMgr->GetSessionManager();
+    if (sessionManager == nullptr)
```

Comment:
Can be simplfied with the `VerifyOrReturn` statement. Also, can this even happen?